crates/rust-analyzer/src/handlers.rs

```diff
@@ -1315,6 +1315,29 @@ pub(crate) fn handle_open_docs(
     Ok(remote.and_then(|remote| Url::parse(&remote).ok()))
 }
 
+pub(crate) fn handle_open_cargo_toml(
+    snap: GlobalStateSnapshot,
+    params: lsp_ext::OpenCargoTomlParams,
+) -> Result<Option<lsp_types::GotoDefinitionResponse>> {
+    let _p = profile::span("handle_open_cargo_toml");
+    let file_id = from_proto::file_id(&snap, &params.text_document.uri)?;
+    let maybe_cargo_spec = CargoTargetSpec::for_file(&snap, file_id)?;
+    if maybe_cargo_spec.is_none() {
+        return Ok(None);
+    }
+
+    let cargo_spec = maybe_cargo_spec.unwrap();
+    let cargo_toml_path = cargo_spec.workspace_root.join("Cargo.toml");
+    if !cargo_toml_path.exists() {
+        return Ok(None);
+    }
+    let cargo_toml_url = to_proto::url_from_abs_path(&cargo_toml_path.clone());
+    let cargo_toml_location =
+        Location::new(cargo_toml_url.clone(), Range::new(Position::new(0, 0), Position::new(0, 0)));
```

Comment:
```suggestion
        Location::new(cargo_toml_url, Range::default();
```

(not sure about the `Default` impl)

Reply:
Also, is the `lsp_types::GotoDefinitionResponse` the right type to return?

Reply:
@lnicola what else should I return? GotoDefinitionResponse seems to be used to jump to a specific part of a file.

Reply:
This is my second time contributing to Rust-Analyzer so it might take me a bit to learn the wheels :|

Reply:
No worries. I was thinking that a custom type or an URL would be better, since we're navigating to a file, not a location.

Reply:
We could maybe use TextDocumentIdentifier and hardcode the location on the client? But maybe in the future we want to open specific parts of the Cargo.toml, and we should keep the client logic as simple as possible, right?